app/scripts/infuse/dimInfuse.factory.js

```diff
@@ -0,0 +1,86 @@
+(function() {
+  'use strict';
+
+  angular.module('dimApp')
+    .factory('infuseService', infuseService);
+
+  infuseService.$inject = [];
+
+  function infuseService() {
+
+    var _data = {
+      source: null,
+      targets: [],
+      infused: 0,
+      view: [],
+      infusable: [],
+      calculate: function() {
+
+        var result = 0;
+        var source = _data.source.primStat.value;
+
+        // Exotics get 70%
+        var multiplier = (_data.source.tier === 'Exotic') ? 0.7 : 0.8;
+
+        for(var i=0;i<_data.targets.length;i++) {
+          var target = _data.targets[i].primStat.value;
+          // if we already have a partial
+          if (result > 0) {
+            var source = result;
```

Comment:
FYI source is not actually being updated here. We're created a newly scoped variable that we aren't doing anything with.